docs/current_docs/api/internals.mdx

```diff
@@ -144,58 +144,152 @@ As this example demonstrates, Dagger object IDs hold the state of their correspo
 
 GraphQL query resolution is triggered only when a leaf value (scalar) is requested. Dagger uses this feature of GraphQL to evaluate workflows "lazily".
 
-An example will make this clearer. First, navigate to [Webhook.site](https://webhook.site), a free online tool that lets you receive and log incoming HTTP requests. Obtain and copy your unique webhook URL, as shown below:
+In practice, this means that if you create a Dagger object (like a container or a directory) but never access its state - either by returning it to the Dagger CLI, reading data from it, or using it in a different Dagger Function - Dagger automatically skips it as part of its optimization process, thinking it's not needed. This is analogous to running a build command but never using the output binary; since the output is never used, Dagger optimizes away the build step.
```

Comment:
Little confused by this statement - mostly the away wording.

> Dagger optimizes away the build step

Reply:
Is "Dagger skips the build step" clearer?

Reply:
Changed